.azure-pipelines/common.yml

```diff
@@ -0,0 +1,115 @@
+parameters:
+  name: ''
+  platform: ''
+  settings: ''
+  py_3_5: false
+  py_3_6: false
+  py_3_7: true
+  py_3_8: false
+  container_name: ''
+  container_startup: ''
+  additional_apt_gets: ''
+  additional_requirements: ''
+
+jobs:
+- job: ${{ format('{0}_{1}', parameters.name, parameters.platform) }}
+
+  timeoutInMinutes: 360
+
+  pool:
+    ${{ if eq(parameters.platform, 'Windows') }}:
+      vmImage: VS2017-Win2016
+    ${{ if eq(parameters.platform, 'Linux') }}:
+      vmImage: Ubuntu 16.04
+
+  strategy:
+    matrix:
+      ${{ if eq(parameters.py_3_5, 'true') }}:
+        Python35:
+          PYTHON_VERSION: '3.5'
+      ${{ if eq(parameters.py_3_6, 'true') }}:
+        Python36:
+          PYTHON_VERSION: '3.6'
+      ${{ if eq(parameters.py_3_7, 'true') }}:
+        Python37:
+          PYTHON_VERSION: '3.7'
+      ${{ if eq(parameters.py_3_8, 'true') }}:
+        Python38:
+          PYTHON_VERSION: '>= 3.8.0a'
+
+  steps:
+    - ${{ if ne(parameters.container_startup, '') }}:
+      - bash: ${{ parameters.container_startup }}
+        displayName: Spin up container
+
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(PYTHON_VERSION)'
+        architecture: 'x64'
+      displayName: Use Python version $(PYTHON_VERSION)
+
+    - ${{ if eq(parameters.platform, 'Linux') }}:
+      - bash: |
+          sudo apt-get -yq update
+          sudo apt-get -yq install libmemcached-dev

```

Comment:
Is this done on every testrun? If yes, this should be part of the image.

Reply:
If `libmemcached-dev` is commonly required by lots of projects, should be no problem to add to the base image. The `apt-get update` line is unnecessary after some other changes I made, so I'll remove it.

Reply:
See my comment above, I didn't realize that azure doesn't allow for custom images

Reply:
Gotcha, the confusion was on my side. For anyone else reading this thread -- we do offer container jobs where you can run in an arbitrary container. I had a tough time making two containers play nice together so switched back to running the main job on the host.

Reply:
Just one other thing to note, do you run your own Ubuntu apt mirror by any chance? A lot of CI projects will use the default one and install the same dependencies over and over again, it might be worth considering this if you're not already.